kitsune/search/v2/base.py

```diff
@@ -0,0 +1,48 @@
+from django.utils import timezone
+
+from elasticsearch_dsl import Document as DSLDocument
+from elasticsearch_dsl import field
+
+
+class SumoDocument(DSLDocument):
+    """Base class with common methods for all the different documents."""
+
+    indexed_on = field.Date()
+
+    @classmethod
+    def prepare(cls, instance):
+        """Prepare an object given a model instance"""
+        obj = cls()
+        doc_mapping = obj._doc_type.mapping
+        fields = [f for f in doc_mapping]
+        fields.remove("indexed_on")
+
+        # Loop through the fields of each object and set the right values
+        for f in fields:
+
+            try:
+                # This will allow child classes to have own methods in the form of prepare_field
+                prepare_method = getattr(obj, "prepare_{}".format(f))
+                value = prepare_method(instance)
+            except AttributeError:
+                value = getattr(instance, f)
+            # Assign values to each field. Assign a dictionary to Objects
+            if isinstance(doc_mapping.resolve_field(f), field.Object):
+                locale = cls.prepare_locale(instance)
+                if locale and value:
+                    obj[f] = {locale: value}
+                # We  populate non-locale object fields here
```

Comment:
I'm not sure it'll be the case that every `field.Object` on an instance with a `.locale` attribute will be an object with multiple locales. I think this will need to be re-jigged a bit for AAQ, but maybe it makes sense for me to do this rather than you writing in the abstract?

Reply:
the comment is just a placeholder for an `else` statement in case we need to populate an Object without a locale. Although in this case the `value` needs to go away from the `if` statement. Let's tackle that when there is a need for it.